universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
@geoffthemedio thanks for writing back.  
@Dilvish-fo thanks for the link.  I will definitely read through that.  Isn't appeal to authority one of those logical fallacy I was warned about 😉 

I should have written, conventions for smart pointer use in C++11.  We are still making the transition to C++11 and wrt to pointer use there are two areas that are still using pre-C++11 conventions: the parser code and GG.  Those two areas of code are still using `new` and `delete` to manage the lifetime of heap based objects. 

Once those are updated then a raw `T*` will explicitly indicate non-ownership of a heap object.  Code can not safely store a raw pointer or allow control to return to the owner and that is also implied by the raw pointer type.  If code does not store raw pointers, then there will not be multiple pointers to the managed objects existing at the same time.

This PR is not about fixing `BuildingTypeManager`.

However, `BuildingTypeManager` is safe and correct to return raw pointer on two accounts.

First, it is static, global, parses the building types once and never de-allocates them.

Second, I checked ~52 uses of its returned pointers and they are correctly used and never stored.

There are two cases where `m_enqueue_location` and `m_location` are stored, but that is an argument for storing/passing those as `shared_ptrs` not the whole `BuildingType`.